edgelet/iotedged/src/lib.rs

```diff
@@ -249,10 +252,12 @@ where
             .context(ErrorKind::Initialize(InitializeErrorReason::Tokio))?;
 
         if let Provisioning::Manual(ref manual) = settings.provisioning() {
-            if manual.device_connection_string() == DEFAULT_CONNECTION_STRING {
-                return Err(Error::from(ErrorKind::Initialize(
-                    InitializeErrorReason::NotConfigured,
-                )));
+            if let ManualAuthMethod::DeviceConnectionString(cs) = manual.authentication_method() {
+                if cs.device_connection_string() == DEFAULT_CONNECTION_STRING {
```

Comment:
This will also become unnecessary when you remove `DEFAULT_CONNECTION_STRING` as I said in my other comment.